### PR TITLE
2024.12.0

### PR DESCRIPTION
**Pull request by frenck**
# DO NOT MERGE

This PR is a work in progress, and used for reviewing purposes.

### CODE REVIEW

File: source/_posts/2024-12-04-release-202412.markdown [hunk 1]
```diff
@@ -0,0 +1,644 @@
+---
+layout: post
+title: "2024.12: Scene you in 2025! 🎄"
+description: "Holidays are coming, time for the last release of the year! View your scenes without immediately activating them. Let your voice assistant fall back to a LLM-based agent, and enjoy a faster voice experience. A revised Integration Quality Scale, new integrations, plus, we've upgraded to Python 3.13! See you in 2025!"
```
Comment (MartinHjelmare) [inline source/_posts/2024-12-04-release-202412.markdown]:
```suggestion
description: "Holidays are coming, time for the last release of the year! View your scenes without immediately activating them. Let your voice assistant fall back to an LLM-based agent, and enjoy a faster voice experience. A revised Integration Quality Scale, new integrations, plus, we've upgraded to Python 3.13! See you in 2025!"
```

File: source/_posts/2024-12-04-release-202412.markdown [hunk 2]
```diff
@@ -0,0 +1,644 @@
...
+- [Translations for units of measurement](#translations-for-units-of-measurement)
+- [We've upgraded to Python 3.13](#weve-upgraded-to-python-313)
+- [Need help? Join the community!](#need-help-join-the-community)
+- [Backward-incompatible changes](#backward-incompatible-changes)
+- [All changes](#all-changes)
+
+## Improved scene editor experience
+
+{% term Scenes %} are a great way to set the mood in your home. They allow you
+to capture the state of your devices at a specific moment and restore them
```
Comment (MartinHjelmare) [inline source/_posts/2024-12-04-release-202412.markdown]:
```suggestion
to capture the state of your devices at a specific moment and restore it
```

File: source/_posts/2024-12-04-release-202412.markdown [hunk 3]
```diff
@@ -0,0 +1,644 @@
...
+One of the common frustrations with editing scenes in Home Assistant is that it
+would immediately activate and apply the scene if you opened the scene editor.
+Not the best experience if you are doing some midnight tinkering and
+accidentally activate a scene that turns on all the lights in the house,
+waking up everyone. 😅
+
+[@karwosts] & [@marcinbauer85] to the rescue! 🦸‍♂️ As of this release, the scene
+editor has two modes. A **review mode** and an **edit mode**.
+
+<img class="no-shadow" alt="The improved scene editor now as a preview mode, which is shown in this screenshot." src="/images/blog/2024-12/scene-review-mode.png"/>
```
Comment (MartinHjelmare) [inline source/_posts/2024-12-04-release-202412.markdown]:
```suggestion
<img class="no-shadow" alt="The improved scene editor now has a preview mode, which is shown in this screenshot." src="/images/blog/2024-12/scene-review-mode.png"/>
```

File: source/_posts/2024-12-04-release-202412.markdown [hunk 4]
```diff
@@ -0,0 +1,644 @@
...
+Curious? Be sure to [join the live stream], hit like 👍, subscribe to our
+channel, and the little bell 🔔 to get notified when we go live! You really
+don't want to miss this one!
+
+But before we get there, let's dive into the features that ship in this release!
+
+[join the live stream]: https://www.youtube.com/watch?v=ZgoaoTpIhm8
+[we started our journey]: /blog/2022/12/20/year-of-voice/
+
+### Let your voice assistant fall back to a LLM-based agent
```
Comment (MartinHjelmare) [inline source/_posts/2024-12-04-release-202412.markdown]:
```suggestion
### Let your voice assistant fall back to an LLM-based agent
```

File: source/_posts/2024-12-04-release-202412.markdown [hunk 5]
```diff
@@ -0,0 +1,644 @@
...
+[organization capabilities]: /blog/2024/04/03/release-20244/#three-new-ways-to-organize
+[the #1 open source project on GitHub]: /blog/2024/11/18/event-wrapup-github-universe-24/#we-are-number-1
+[the live stream on 19 December]: https://youtube.com/live/ZgoaoTpIhm8
+[the WTH announcement blog]: /blog/2024/12/01/announcing-month-of-what-the-heck/
+
+<!--more-->
+
+- [Improved scene editor experience](#improved-scene-editor-experience)
+- [Voice](#voice)
+  - [Let your voice assistant fall back to a LLM-based agent](#let-your-voice-assistant-fall-back-to-a-llm-based-agent)
```
Comment (MartinHjelmare) [inline source/_posts/2024-12-04-release-202412.markdown]:
```suggestion
  - [Let your voice assistant fall back to an LLM-based agent](#let-your-voice-assistant-fall-back-to-an-llm-based-agent)
```

File: source/_posts/2024-12-04-release-202412.markdown [hunk 6]
```diff
@@ -0,0 +1,644 @@
...
+
+<img class="no-shadow" alt="Dialog showing the Assist pipeline configuration, showing the new option to prefer handling commands locally." src="/images/blog/2024-12/llm-fallback-configuration.png"/>
+
+This allows you to mix the pros of both worlds with almost none of the cons.
+
+Specific known commands will be processed locally and extremely fast, and the
+power of an LLM will only be used for more complex queries that Home Assistant
+does not natively understand.
+
+<lite-youtube videoid="vThoxRIxHyI" videotitle="Let your voice assistant fall back to a LLM-based agent" posterquality="maxresdefault"></lite-youtube>
```
Comment (MartinHjelmare) [inline source/_posts/2024-12-04-release-202412.markdown]:
```suggestion
<lite-youtube videoid="vThoxRIxHyI" videotitle="Let your voice assistant fall back to an LLM-based agent" posterquality="maxresdefault"></lite-youtube>
```

File: source/_posts/2024-12-04-release-202412.markdown [hunk 7]
```diff
@@ -0,0 +1,644 @@
...
+a non-matching sentence could take more than 15 seconds to process!
+
+This release adds a lot of love to this matching logic; we completely
+reengineered the way we match sentences. On top of that, language leaders spent
+some time reducing the complexity of their sentences. Some languages saw a 99%
+reduction in the number of possible combinations!
+
+The results speak for themselves:
+
+<lite-youtube videoid="RH69Dh6coMM" videotitle="A faster voice experience with Our default conversation agent" posterquality="maxresdefault"></lite-youtube>
```
Comment (MartinHjelmare) [inline source/_posts/2024-12-04-release-202412.markdown]:
```suggestion
<lite-youtube videoid="RH69Dh6coMM" videotitle="A faster voice experience with our default conversation agent" posterquality="maxresdefault"></lite-youtube>
```

File: source/_posts/2024-12-04-release-202412.markdown [hunk 8]
```diff
@@ -0,0 +1,644 @@
...
+  Adds support for [Acaia coffee scales] to Home Assistant.
+- **[Music Assistant]**, added by [@jozefKruszynski]  
+  The integration for [Music Assistant](https://music-assistant.io/) is making
+  its way into Home Assistant! 🎉
+- **[NASweb]**, added by [@nasWebio]  
+  Brings in support for the NASweb automation system to Home Assistant.
+- **[Nord Pool]**, added by [@gjohansson-ST]  
+  Pulls in latest energy prices from the Nord Pool energy market as sensors.
+- **[Sky Remote]**, added by [@dunnmj]  
+  Control your [Sky] box using automations on from your Home Assistant
```
Comment (MartinHjelmare) [inline source/_posts/2024-12-04-release-202412.markdown]:
```suggestion
  Control your [Sky] box using automations or from your Home Assistant
```

File: source/_posts/2024-12-04-release-202412.markdown [hunk 9]
```diff
@@ -0,0 +1,644 @@
...
+  energy-monitoring sensors for Tapo devices. It also exposes new entities like
+  a motion sensor switch, a child lock switch for thermostats, and the last
+  water leak alert timestamp.
+  Thanks to [@sdb9696], [@rytilahti], and [@ryenitcher]!
+- [@starKillerOG] added selector entities for bitrate, framerate, and vehicle
+  tone to the [Reolink integration]. Additionally, a tilt position sensor for
+  PTZ cameras has been added. Thanks!
+- [@fwestenberg] updated the [Stookwijzer integration], with some new sensors,
+  right on time for the winter season. Nice!
+- The [Habitica integration] now provides action for quest handling, scoring
```
Comment (MartinHjelmare) [inline source/_posts/2024-12-04-release-202412.markdown]:
```suggestion
- The [Habitica integration] now provides actions for quest handling, scoring
```

File: source/_posts/2024-12-04-release-202412.markdown [hunk 10]
```diff
@@ -0,0 +1,644 @@
...
+
+[@gjohansson-ST]: https://github.com/gjohansson-ST
+[#131427]: https://github.com/home-assistant/core/pull/131427
+
+{% enddetails %}
+
+{% details "History Stats" %}
+
+After this change, `history_stats` in `time` or `ratio` mode will only count
+definitively known time-based on recorded data.
```
Comment (MartinHjelmare) [inline source/_posts/2024-12-04-release-202412.markdown]: Not sure what this sentence wants to say. I don't understand it. Sounds like some words should change at least.

File: source/_posts/2024-12-04-release-202412.markdown [hunk 11]
```diff
@@ -0,0 +1,644 @@
...
+
+Prior to this change, if the start of the time window was earlier than the
+oldest datapoint in the recorder, `history_stats` would count time using the
+assumption that an entity was in the first state it found since the beginning
+of time. This could lead to wild overcalculations when a switch was turned on
+after a long period of inactivity, and `history_stats` would assume it has
```
Comment (MartinHjelmare) [inline source/_posts/2024-12-04-release-202412.markdown]:
```suggestion
after a long period of inactivity, and `history_stats` would assume it had
```

File: source/_posts/2024-12-04-release-202412.markdown [hunk 12]
```diff
@@ -0,0 +1,644 @@
...
+- `network_tx`
+- `status`
+- `system_temp`
+- `uptime`
+- `volume_percentage_used`
+- `volume_size_free`
+- `volume_size_total`
+- `volume_size_used`
+
+These extra state attributes were deprecated in Home Assistant 2024.6 and have
```
Comment (MartinHjelmare) [inline source/_posts/2024-12-04-release-202412.markdown:528]: Are the listed items both sensors and extra attributes?

File: source/_posts/2024-12-04-release-202412.markdown [hunk 13]
```diff
@@ -0,0 +1,644 @@
...
+
+[@joostlek]: https://github.com/joostlek
+[#131754]: https://github.com/home-assistant/core/pull/131754
+[#131758]: https://github.com/home-assistant/core/pull/131758
+
+{% enddetails %}
+
+{% details "Statistics" %}
+
+Previously the attributes were only provided when they had a non `None` value.
```
Comment (MartinHjelmare) [inline source/_posts/2024-12-04-release-202412.markdown]: Would be good to mention what attributes we are talking about.

Comment (MartinHjelmare) [reply source/_posts/2024-12-04-release-202412.markdown]: At minimum:
```suggestion
Previously the attributes of the entity states of the Statistics integration were only provided when they had a non `None` value.
```

File: source/_posts/2024-12-04-release-202412.markdown (hunk 10 above)
Comment (MartinHjelmare) [reply source/_posts/2024-12-04-release-202412.markdown]: Maybe like this?
```suggestion
definitively known time-based recorded data.
```

File: source/_posts/2024-12-04-release-202412.markdown (hunk 12 above)
Comment (frenck) [reply source/_posts/2024-12-04-release-202412.markdown:528]: Extra attributes as it seems.

Comment (MartinHjelmare) [reply source/_posts/2024-12-04-release-202412.markdown:528]: Above the list it says "the following sensors".

File: source/_posts/2024-12-04-release-202412.markdown [hunk 14]
```diff
@@ -0,0 +1,645 @@
...
+[the #1 open source project on GitHub]: /blog/2024/11/18/event-wrapup-github-universe-24/#we-are-number-1
+[the live stream on 19 December]: https://youtube.com/live/ZgoaoTpIhm8
+[the Open Home Foundation]: https://www.openhomefoundation.org/
+[the WTH announcement blog]: /blog/2024/12/01/announcing-month-of-what-the-heck/
+
+<!--more-->
+
+- [Improved scene editor experience](#improved-scene-editor-experience)
+- [Voice](#voice)
+  - [Let your voice assistant fall back to an LLM-based agent](#let-your-voice-assistant-fall-back-to-a-llm-based-agent)
```
Comment (MartinHjelmare) [inline source/_posts/2024-12-04-release-202412.markdown]:
```suggestion
  - [Let your voice assistant fall back to an LLM-based agent](#let-your-voice-assistant-fall-back-to-an-llm-based-agent)
```

File: source/_posts/2024-12-04-release-202412.markdown [hunk 15]
```diff
@@ -0,0 +1,645 @@
...
+
+Home Assistant 2024.12! 🎄
+
+Holidays are coming, and it is time for the last release of the year! 🎉
+
+**2024** has been a crazy year for Home Assistant. Not just in terms of features
+like [drag-and-drop dashboards], [organization capabilities] like labels, and
+the countless improvements to our voice efforts. But also the founding of
+[the Open Home Foundation], and even more in the growth of our lovely community
+as we risen up to become [the #1 open source project on GitHub]! ❤️
```
Comment (MartinHjelmare) [inline source/_posts/2024-12-04-release-202412.markdown]:
```suggestion
as we've risen up to become [the #1 open source project on GitHub]! ❤️
```

File: source/_posts/2024-12-04-release-202412.markdown [hunk 16]
```diff
@@ -0,0 +1,645 @@
...
+([@tsvi] - [#130456]) ([documentation](/integrations/jewish_calendar))
+
+[@tsvi]: https://github.com/tsvi
+[#130456]: https://github.com/home-assistant/core/pull/130456
+
+{% enddetails %}
+
+{% details "QNAP" %}
+
+The following QNAP sensors will no longer have extra state attributes:
```
Comment (frenck) [inline source/_posts/2024-12-04-release-202412.markdown]:
```suggestion
The QNAP sensors will no longer have the following extra state attributes:
```

File: source/_posts/2024-12-04-release-202412.markdown (hunk 12 above)
Comment (frenck) [reply source/_posts/2024-12-04-release-202412.markdown:528]: Fixed

**Pull request merged**